### DISCUSSION TITLE
Debian 12 - How to use Wayfire?

### DISCUSSION THREAD

**Discussion by liszca** [WayfireWM/wayfire#2067] [General]
Having Debian 12 installed I tried to get it running. As wayland is already working out of the box installing "wayfire" with apt was easy. 

Then at the Login Window I choose Wayfire instead of GNOME but got set back to Login screen. 

After looking around I thought putting "wayfire.ini" into my config folder would help but still I got set back to Login. 

What is missing for it to work under Debian 12? Saw it working on the Raspberry PI 4 and was impressed. But how to do it on x86?

**Comment by ammen99** [comment]
That should be enough, but it sounds like Wayfire is crashing at startup. Can you try launching it nested inside X11 or another Wayland compositor? That way wayfire will run nested in a window for testing. If that works, try to run it from a tty and redirect stdout/stderr to a log file to see what the error is.

**Reply by liszca** [reply]
how would  I launch it under wayland gnome?

**Reply by ammen99** [reply]
Just type `wayfire` in a terminal.

**Reply by liszca** [reply]
In a VM I get "[src/main.cpp:268] Failed to get DRM file Descriptor" (Debian 12).

Under Ubuntu 23.10 (Wayland) an undecorated window appears
Terminal throws an error: [types/wlr_cursor.c:875] Cannot map device "wayland-keyboard-seat0".

**Reply by soreau** [reply]
Did you also install something like wf-shell so it has something to display in the window? Otherwise it's just a black window because wayfire basically just serves clients. So you'll have to run a shell or other client, in addition to starting wayfire.

**Reply by liszca** [reply]
no nothing else I installed. Cannot find a package named "wf-shell" (Debian 12 & Ubuntu 23.10).

**Reply by soreau** [reply]
For example, if you run wayfire and in the output, it says something like "Using socket name wayland-2", then you should be able to start wayland capable programs in wayfire with `WAYLAND_DISPLAY=wayland-2 <app>`. Though the better way to do this is to simply add an entry to the config file it's using, usually `~/.config/wayfire.ini`. There you can modify it as outlined in the wayfire wiki, to automatically start things for you when you start wayfire. You can install waybar and swaybg from the repos most likely, and autostart those for a custom shell.

**Reply by liszca** [reply]
So I started over, now I have a Debian 12 without GUI. Setting systemd to Graphical target and installing wayfire wasn't enough. Is the login manager missing?

Also I set my console to framebuffer mode, not sure if that can be a problem. But errormessages look the same with or without framebuffer

In general my goal is to replicate the raspberry pi wayfire desktop.

**Reply by liszca** [reply]
After installing GDM3 login to a wayfire session worked.

But putting wafire.ini into position didn't change anything.
From the wayfire.ini I did expect to have some panels, but nothing did show up.
